src/main/java/org/akhq/repositories/RecordRepository.java

```diff
@@ -61,6 +72,32 @@
     @Value("${akhq.topic-data.poll-timeout:1000}")
     protected int pollTimeout;
 
+
+    public Optional<Record> getLastRecord(String clusterId, String topicName) throws ExecutionException, InterruptedException {
```

Comment:
```java
    public Optional<Record> getLastRecord(String clusterId, String topicName) throws ExecutionException, InterruptedException {
        KafkaConsumer<byte[], byte[]> consumer = kafkaModule.getConsumer(clusterId);
        Topic topic = topicRepository.findByName(clusterId, topicName);

        List<TopicPartition> topicPartitions = topic
                .getPartitions()
                .stream()
                .map(partition -> new TopicPartition(partition.getTopic(), partition.getId()))
                .collect(Collectors.toList());

        consumer.assign(topicPartitions);
        AtomicReference<ConsumerRecord<byte[], byte[]>> latestRecord = new AtomicReference<>();

        consumer
            .endOffsets(consumer.assignment())
            .forEach((topicPartition, offset) -> {
                consumer.seek(topicPartition, Math.max(0, offset - 3));
            });

        AtomicLong maxTimestamp = new AtomicLong();
        this.poll(consumer).forEach(record -> {
            if (record.timestamp() > maxTimestamp.get()) {
                maxTimestamp.set(record.timestamp());
                latestRecord.set(record);
            }
        });

        consumer.close();
        return Optional.ofNullable(latestRecord.get()).map(record -> newRecord(record, clusterId));
    }
```
I need to change to this to make it work with topic having transactions.
Also I poll only one time except one time per topic for performance.

Reply:
Hey @tchiotludo , Why should it be offset - 3?

Reply:
-2 must be sufficient I think. 
Kafka transaction commit an offset that the consumer can't read, if you try with -1 you will never get the last message

Reply:
how to determine if topic is transactional? We can add an if check for that - if transactional, give offset -2 else 1

Reply:
You can't, it's only known by the producer and a topic can contains transactions or not, no way to know it

Reply:
then we can check both offsets and return a value if found

Reply:
Could you give example for opening the linked record in a view? Will try to incorporate that change as well